src/components/Inventory/AddInventory.js

```diff
@@ -0,0 +1,331 @@
+import { useRef, useState, useEffect } from "react";
+import "./AddInventory.scss";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
+import BackIcon from "../../assets/icons/arrow_back-24px.svg";
+import ErrorIcon from "../../assets/icons/error-24px.svg";
+
+const AddInventory = () => {
+  const navigate = useNavigate();
+
+  const initialState = useRef({
+    item_name: "",
+    description: "",
+    category: "",
+    status: "In Stock", // default status
+    quantity: "",
+    warehouse_name: "",
+  });
+
+  // State to keep track of form data
+  const [formData, setFormData] = useState(initialState.current);
```

Comment:
Since you are putting a state to this form. With useRef you will still see re-renderings happen on multiple key strokes. Its fine because its just the way forms tend to work in React. Libraries like Formik, React-hooks-form will help